table-to-blt.mjs

```diff
@@ -0,0 +1,147 @@
+*
+* The format of the table is that generated by a WBS rank-order
+* control. The code assumes that the title attribute on each
+* table cell includes both the rank information and the candidate.
+*
+* The rank-order control allows voters to leave candidates "Unranked".
+*
+* Arguments:
+* - HTML file with a table with class including "results"; the code picks the first one
+*   because this is what WBS forms produce.
+* - The number of seats available for election
+* - Optional: The title that should appear in the BLT file (not semantically important).
+*
+* For information about the BLT file format, see:
+* https://github.com/Conservatory/openstv/blob/master/openstv/Help.html
+*
+* Assumptions:
+* - Each ballot has a weight of 1.
+* - OpenSTV says: You don't need to rank all of the candidates on each
+*                 ballot, but you cannot skip rankings and you 
+*                 cannot give two or more candidates the same ranking."
+*   Therefore, we ignore ballots with gaps and ballots where two or
+*   more candidates have the same ranking. However, we report these phenomena.
+* - OpenSTV says: "OpenSTV only accepts valid ballots, and an empty
+*   ballot is valid." Therefore we preserve empty ballots.
+*/
+
+import { readFile } from 'node:fs/promises';
+import { JSDOM } from 'jsdom';
+
+async function main(file, nbseats, electionname) {
+    const html = await readFile(file, { encoding: 'utf8'});
+    let dom = new JSDOM(html);
+    generateBLT(dom, nbseats, electionname);
+}
+
+function generateBLT (dom, nbseats, electionname) {
+  // NOTE: The first column is the name of the voter.
+  let table = dom.window.document.querySelector("table")
+
+  // Ignore first column in determining number of candidates.
+  const nbcandidates = table.querySelector("tr").querySelectorAll('td,th').length - 1;
+  let rows = table.querySelectorAll("tr");
+
+  // candidates is the list of candidate names, taken from the top (zeroth) row of the table.
+  const candidates = Array.from(rows[0].querySelectorAll('th')).map(s => s.innerHTML).slice(1) ;
+    
+  // Create ballots
+  const ballots = generateBallots(dom, rows, candidates);
+  
+  // Start output in BLT format
+  // First row is number of candidates and number of seats
+  console.log(nbcandidates + " " + nbseats);
+  // List of ballots, each with weight "1" and ending with "0"
+  console.log(ballots.map(s => "1 " + s.join(" ") + " 0").join("\n"));
+  // Zero separator
+  console.log("0");
+  // Names of candidates, ignoring the first column.
+  for (let i = 1; i <= nbcandidates; i++) {
+    console.log("\"Candidate " + i + "\"");
+  }
+  // Election name
+  console.log("\"" + electionname + "\"");
+}
+
+function generateBallots(dom, rows, candidates) {
+    let ballots = [];
+    // Handle rows. Ignore top (zeroth) row since that is candidate names.
+    for (let i = 1; i < rows.length; i++) {
+        // Handle cells in the row.
+        let cells = rows[i].querySelectorAll('td');
```

Comment:
when walking an array, I would usually use
```suggestion
        for (const row of rows) {
```
IOW, `i` is a distraction when the real focus of the loop is `rows[i]` - (here, and in other similar loops).

In this case, since you're skipping the first row, it would be 
```suggestion
        for (const row of rows.slice(1)) {
```